# HG changeset patch
# User Rik <rik@octave.org>
# Date 1388781377 28800
#      Fri Jan 03 12:36:17 2014 -0800
# Branch stable
# Node ID 48dafd7398408beeb2c9a9b6077ace9c6e82adb3
# Parent  2fbfeacf43f4e85b71b327f1b73d32e5769ed531
cov.m: Document intentional Matlab incompatibility (bug #40751).

cov.m: Document intentional Matlab incompatibility (bug #40751).

diff --git a/scripts/statistics/base/cov.m b/scripts/statistics/base/cov.m
--- a/scripts/statistics/base/cov.m
+++ b/scripts/statistics/base/cov.m
@@ -50,16 +50,23 @@
 ## @table @asis
 ## @item 0:
 ##   normalize with @math{N-1}, provides the best unbiased estimator of the
 ## covariance [default]
 ##
 ## @item 1:
 ##   normalize with @math{N}, this provides the second moment around the mean
 ## @end table
+##
+## @sc{matlab} compatibility: Octave always computes the covariance matrix.
+## For two inputs, however, @sc{matlab} will calculate
+## @code{cov (@var{x}(:), @var{y}(:))} whenever the number of elements in
+## @var{x} and @var{y} are equal.  This will result in a scalar rather than
+## a matrix output.  Code relying on this odd definition will need to be
+## changed when running in Octave.
 ## @seealso{corr}
 ## @end deftypefn
 
 ## Author: KH <Kurt.Hornik@wu-wien.ac.at>
 ## Description: Compute covariances
 
 function c = cov (x, y = [], opt = 0)
 
